ext/dom/element.c

```diff
@@ -2030,6 +2031,11 @@ PHP_METHOD(Dom_Element, rename)
 			}
 			goto cleanup;
 		}
+
+		/* If we currently have a template but the new element type won't be a template, then throw away the templated content. */
+		if (is_currently_html_ns && xmlStrEqual(nodep->name, BAD_CAST "template") && !xmlStrEqual(localname, BAD_CAST "template")) {
+			php_dom_remove_templated_content(php_dom_get_private_data(intern), nodep);
+		}
 	}
```

Comment:
The rename method is a PHP-specific addition. This case is about a template element that gets renamed to another name. Because only template elements have templated content, keeping the templated content around after a rename is nonsensical.

Reply:
I think I would rather have this throw an Exception then? Because this feels like it can be surprising/confusing?

Reply:
That's fair, that is the more safe option because we could introduce behavior later, but we can't take away behaviour.